geomstats/geometry/spd_matrices.py

```diff
@@ -644,6 +675,31 @@ def log(self, point, base_point):
         log_at_id = logmh(point_near_id)
         return Matrices.mul(sqrt_base_point, log_at_id, sqrt_base_point)
 
+    def squared_dist(self, point_a, point_b):
+        """Compute the Affine Invariant squared distance.
+
+        Compute the Riemannian squared distance between point_a and point_b.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., n, n]
+            Point.
+        point_b : array-like, shape=[..., n, n]
+            Point.
+
+        Returns
+        -------
+        squared_dist : array-like, shape=[...]
+            Riemannian squared distance.
+
+        Notes
+        -----
+        Use of `abs` in the output prevents nan when calling
+        `sqrt` in very small negative outputs (e.g. -1e-16).
+        """
+        gen_eigvals = generalized_eigenvalues(point_a, point_b)
+        return gs.abs(gs.sum(gs.log(gen_eigvals) ** 2, axis=-1))
```

Comment:
Have you checked this happens often? What would be the cases? Distance of a point against itself?

Reply:
I haven't actually, I just copied it (code and motivation) from the Bures Wasserstein squared distance down below. I guessed that if that distance had this issue, the Affine Invariant would too.

Reply:
All changes look good. It's nice to get a feel for the style for potential future contributions.

Small note: I mentioned the new procedure is more stable, not more accurate (probably related though). If you generate the spd matrices by generating n-by-b matrix `A` with randn and taking `SPD = A @ A.t()`, a non-trivial proportion of times the original implementation will generate Nan's, while the new one won't. In the script I shared, SPD matrices are sampled with the geomstats built in method and the original method doesn't generate NaN's.

Reply:
I've tested without `abs` and it seems to work properly. For now I'll merge without it, and if we have problems later I'll add it back in (I prefer to have the less "hacking" possible in the code if that works fine).

Thanks for the clarification on stable vs accurate. I agree with you that they're probably very much correlated.